integration/kubernetes/vfio.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_DIR}/../../.ci/lib.sh"
+
+SYSCONFIG_FILE="/etc/kata-containers/configuration.toml"
+
+trap cleanup EXIT
+cleanup() {
+	rm -rf "${SYSCONFIG_FILE}"
+	${SCRIPT_DIR}/cleanup_env.sh
+}
+
+setup_configuration_file() {
+	local image_type="$1"
+	local machine_type="$2"
+	local hypervisor="$3"
+	local sandbox_cgroup_only="$4"
+
+	local default_config_file="/usr/share/defaults/kata-containers/configuration.toml"
+	local qemu_config_file="/usr/share/defaults/kata-containers/configuration-qemu.toml"
+	local clh_config_file="/usr/share/defaults/kata-containers/configuration-clh.toml"
+	local image_file="/usr/share/kata-containers/kata-containers.img"
+	local initrd_file="/usr/share/kata-containers/kata-containers-initrd.img"
+
+	mkdir -p $(dirname "${SYSCONFIG_FILE}")
+
+	if [ "${hypervisor}" = "qemu" ]; then
+		config_file="${qemu_config_file}"
+	elif [ "${hypervisor}" = "cloud-hypervisor" ]; then
+		config_file="${clh_config_file}"
+	fi
+
+	if [ -f "${config_file}" ]; then
+		cp -a "${config_file}" "${SYSCONFIG_FILE}"
+	elif [ -f "${default_config_file}" ]; then
+		# Check if path contains the hypervisor name
+		grep "^path" "${default_config_file}" | grep -q "${hypervisor}"
```

Comment:
This looks weird. I guess it will dtrt as we're running `set -e`, but I'd consider protecting against an exit and calling `die` with a helpful message here.